src/include/platform/DeviceInstanceInfoProvider.h

```diff
@@ -29,6 +29,19 @@ class DeviceInstanceInfoProvider
     DeviceInstanceInfoProvider()          = default;
     virtual ~DeviceInstanceInfoProvider() = default;
 
+    /**
+     * @brief Obtain the Device Name from the device's factory data.
+     *
+     * @param[out] buf Buffer to copy string.
+     *                 On CHIP_NO_ERROR return from this function this buffer will be null-terminated.
+     *                 On error CHIP_ERROR_BUFFER_TOO_SMALL there is no guarantee that buffer will be null-terminated.
+     * @param[in] bufSize Size of data, including the null terminator, that can be written to buf.
+     *                    This size should be +1 higher than maximum possible string.
+     * @returns CHIP_NO_ERROR on success, or another CHIP_ERROR from the underlying implementation
+     *          if access fails.
+     */
+    virtual CHIP_ERROR GetDeviceName(char * buf, size_t bufSize) = 0;
```

Comment:
Please use MutableCharSpan as a type, which gets resized to "actual" name for this since it's a new API. Keeping the prior error-prone and zero-termination-requiring API pattern is not needed.

Reply:
This will be addressed in #28462. I am canceling this PR.

Reply:
@sharadb-amazon Whenever you close a PR that someone left comments on in favor of a different PR, please move the comments to that other PR.

Reply:
Done